lib/req/steps.ex

```diff
@@ -2333,6 +2335,92 @@ defmodule Req.Steps do
     inspect(str, printable_limit: :infinity)
   end
 
+  @doc """
+  Expect that response matches the given status.
+
+  This step ensures the HTTP response has the given expected status, otherwise it
+  returns `Req.UnexpectedResponseError`.
+
+  ## Request Options
+
+    * `:expect` - the expected HTTP response status. Can be one of the following:
+
+        * integer
+        * range
+        * list of integers/ranges
+
+  > #### Order Matters! {: .info}
+  >
+  > By default, `expect/1` runs AFTER `retry/1`, `redirect/1`, `decompress_body/1`,
+  > and `decode_body/1` steps.
+  >
+  > This means that, for example, HTTP 503 error would be first retried,
+  > HTTP 307 redirect would be first followed, and the response body
+  > would be first decompressed and decoded before checking for expected HTTP status.
+  > If this is undesirable, re-arrange or disable and manually run given steps.
+
+  > #### Sensitive Response Data {: .warning}
+  >
+  > This steps returns `Req.UnexpectedResponseError` which contains full `Req.Response`.
+  > Since response headers/body can contain sensitive data, be careful about raising
+  > this error and automatically logging it, sending to exception trackers, etc.
+
+  ## Examples
+
+      iex> resp = Req.get!("https://httpbin.org/status/200", expect: 200)
+      iex> resp.status
+      200
+
+      iex> Req.get!("https://httpbin.org/status/404", expect: 200..299)
```

Comment:
```suggestion
      iex> Req.get!("https://httpbin.org/status/404", expect: :ok)
```

could be an atom representing a given config as well? 😅